compiler/src/iree/compiler/Codegen/Common/DecomposeBatchMmt4DOps.cpp

```diff
@@ -16,6 +16,48 @@ namespace iree_compiler {
 
 namespace {
 
+// Returns true if:
+//    1. `genericOp` is element-wise with all identity indexing maps
+//    2. `genericOp` has only one input and one output with the same shape
+static bool isElementWiseIdentity(linalg::GenericOp genericOp) {
+  if (genericOp.getNumDpsInputs() != 1 || genericOp.getNumDpsInits() != 1 ||
+      !linalg::isElementwise(genericOp) ||
+      !llvm::all_of(genericOp.getIndexingMapsArray(),
+                    [](AffineMap map) { return map.isIdentity(); })) {
+    return false;
+  }
+  auto inputType = genericOp->getOperandTypes()[0].dyn_cast<RankedTensorType>();
+  auto resultType = genericOp->getResultTypes()[0].dyn_cast<RankedTensorType>();
+  if (inputType.getShape() != resultType.getShape()) {
+    return false;
+  }
```

Comment:
The check is not needed (and it not correct). Because they must match if all the affine maps are identity.

It is not correct because they could be all dynamic sizes. In this case, there could be transpose ops involved.